perun-rpc/src/main/java/cz/metacentrum/perun/rpc/Api.java

```diff
@@ -244,35 +268,24 @@ else if (extLogin == null && req.getAttribute("SSL_CLIENT_VERIFY") != null && ((
 				try {
 					altNames = certs[0].getSubjectAlternativeNames();
 					if (altNames != null) {
-						for (List<?> entry: altNames) {
+						for (List<?> entry : altNames) {
 							if (((Integer) entry.get(0)) == 1) {
 								emails = (String) entry.get(1);
 							}
 						}
 					}
 				} catch (CertificateParsingException e) {
-					log.error("Error during parsing certificate {}", certs);
+					log.error("Error during parsing certificate {}", Arrays.asList(certs));
 				}
 
 				additionalInformations.put("mail", emails);
-
-				// Get organization from the certificate
-				String oRegExpPattern = "(o|O)(\\s)*=([^+,])*";
-				Pattern oPattern = Pattern.compile(oRegExpPattern);
-				Matcher oMatcher = oPattern.matcher(certs[0].getSubjectX500Principal().getName());
-				if (oMatcher.find()) {
-					String[] org = oMatcher.group().split("=");
-					if (org[1] != null && !org[1].isEmpty()) {
-						additionalInformations.put("o", org[1]);
-					}
-				}
 			}
 		}
 
 		// Read all headers and store them in additionalInformation
-		String headerName = "";
-		for(Enumeration<String> headerNames = req.getHeaderNames(); headerNames.hasMoreElements();){
-			headerName = (String)headerNames.nextElement();
+		String headerName;
+		for (Enumeration<String> headerNames = req.getHeaderNames(); headerNames.hasMoreElements(); ) {
```

Comment:
I checked and if needed, ``referer`` header is taken from servlet request. Also servlet request is present in deserializer. On side of GUI and WUI, there doesn't seem to be any usage of ``additionalAttributes`` outside federated ones. So I believe this change is safe. But as you noted, before deployment custom CGI scripts must be checked on each instance (for usage headers and perun session attributes).